js/binancetr.js

```diff
@@ -0,0 +1,308 @@
+            const filters = this.safeValue (market, 'filters');
+            const filter0 = this.safeValue (filters, 0);
+            const filter2 = this.safeValue (filters, 2);
+            const permissions = this.safeValue (market, 'permissions');
+            const minPrice = this.safeValue (filter0, 'minPrice');
+            const maxPrice = this.safeValue (filter0, 'maxPrice');
+            const minQty2 = this.safeValue (filter2, 'minQty');
+            const maxQty2 = this.safeValue (filter2, 'maxQty');
+            const baseId = this.safeString (market, 'baseAsset');
+            const quoteId = this.safeString (market, 'quoteAsset');
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const spotPer = permissions.indexOf ('SPOT') > -1;
+            const marginPer = permissions.indexOf ('MARGIN') > -1;
+            const marginEnable = (this.safeInteger (market, 'marginTradingEnable')) === '1' ? true : false;
+            const spotEnable = (this.safeInteger (market, 'spotTradingEnable')) === '1' ? true : false;
+            const symbol = base + '/' + quote;
+            const precision = {
+                'price': this.safeInteger (market, 'quotePrecision'),
+                'amount': this.safeInteger (market, 'basePrecision'),
+            };
+            result.push (this.extend (this.fees['trading'], {
+                'info': market,
+                'id': id,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'type': 'spot',
+                'spot': spotPer,
+                'margin': marginPer && marginEnable,
+                'active': spotEnable,
+                'precision': precision,
+                'limits': {
+                    'amount': {
+                        'min': this.parseNumber (minQty2),
+                        'max': this.parseNumber (maxQty2),
+                    },
+                    'price': {
+                        'min': this.parseNumber (minPrice),
+                        'max': this.parseNumber (maxPrice),
+                    },
+                    'cost': {
+                        'min': undefined, // this.parseNumber (minQty * minPrice),
+                        'max': undefined, // this.parseNumber (maxQty * maxPrice),
+                    },
+                },
+            }));
+        }
+        return result;
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {
+            'symbol': this.marketId (symbol),
+        };
+        if ((limit !== 5) && (limit !== 10) && (limit !== 20) && (limit !== 50) && (limit !== 100) && (limit !== 500)) {
+            throw new BadRequest (this.id + ' fetchOrderBook() limit argument must be undefined, 5, 10, 20, 50, 100 or 500, default is 100');
```

Comment:
It will throw an exception here if `limit` is `undefined`.

Reply:
Well observed. Fixed.